var/spack/repos/builtin/packages/readline/package.py

```diff
@@ -45,4 +45,6 @@ class Readline(AutotoolsPackage):
     patch('readline-6.3-upstream_fixes-1.patch', when='@6.3')
 
     def build(self, spec, prefix):
-        make('SHLIB_LIBS=-lncursesw')
+        options = ['SHLIB_LIBS={0}'.format(spec['ncurses'].libs.ld_flags)]
```

Comment:
True. I'm not sufficiently observant this morning and just dismissed this change in the output of `otool -L`.

I've implemented your suggestion, and now I believe that we DO get the desired result:
```sh
$ otool -L libreadline.dylib
libreadline.dylib:
	<path>/spack/opt/spack/darwin-elcapitan-x86_64/clang-8.0.0-apple/readline-7.0-mvh3ubvrbid3gdjzs4od5gch72rcxius/lib/libreadline.7.dylib (compatibility version 7.0.0, current version 7.0.0)
	<path>/spack/opt/spack/darwin-elcapitan-x86_64/clang-8.0.0-apple/ncurses-6.0-6njlipmhhkb7vulrx3ftc3af7aryq3h7/lib/libncursesw.6.dylib (compatibility version 6.0.0, current version 6.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```